mypy/checkexpr.py

```diff
@@ -2139,15 +2141,13 @@ def infer_literal_expr_type(self, value: LiteralValue, fallback_name: str) -> Ty
            the comments in Instance's constructor for more details.
         """
         typ = self.named_type(fallback_name)
+        typ.set_line(context)
```

Comment:
I think that we can write a test case for this, maybe even inside `check-columns.test`